text_collector_examples/pacman.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+#
+#
+# Description: Expose metrics from pacman updates
+# If installed The bash script *checkupdates*, included with the
+# *pacman-contrib* package, is used to calculate the number of pending updates.
+# Otherwise *pacman* is used for calculation.
+#
+# Author: Sven Haardiek <sven@haardiek.de>
+
+set -o errexit
+set -o nounset
+
+if [ -x /usr/bin/checkupdates ]
+then
+    updates=$(/usr/bin/checkupdates | wc -l)
+    cache=0
+else
+    updates=$(/usr/bin/pacman -Qu | wc -l)
+    cache=1
+fi
+
+echo "# HELP updates_pending number of pending updates, derived live or from cache"
+echo "# TYPE updates_pending gauge"
+echo "updates_pending{package_manager=\"pacman\",from_cache=\"$cache\"} $updates"
```

Comment:
Two suggestions.

I would make the metric name `pacman_updates_pending`, as there are no other package managers supported by this script.

I'd also would make the cache a separate boolean metric, `pacman_updates_pending_from_cache 0/1`.

Reply:
Splitted and renamed the metric.